### ISSUE TITLE
Changing contentHeight while dragging causes event to be dropped as if external

### ISSUE THREAD

**Issue by kaeverens** [fullcalendar/fullcalendar#6561]
### Reduced Test Case

https://demo.verens.com/fullcalendar-bug/bug.html

### Bug Description

drag the event. if you drag and drop in under 1 second, then it will succeed and an alert will show. if your drag is longer than one second, then the eventDrop will fail to run (and the alert will not be shown)

what I expected was for the drag to succeed in all cases.

if you read the example, I've set up an `eventDidMount` that runs `calendar.setOption('contentHeight', 650);` after 1000ms. this event is triggered each time the event is dragged over a valid slot, causing the code to run.

this changes the context so that when the mouse is finally released, FC believes that the drop is happening in a *different calendar*.

this `if` statement around line 11100 in `main.js` is what is failing:

![image](https://user-images.githubusercontent.com/2878523/134688185-21e66d76-3fdf-4a17-850c-b43d09e05d77.png)

my workaround was to avoid resetting the calendar height until after I'm certain the drop is complete (I check to see if info.el has a class "fc-event-dragging")

I think the fact that I need that workaround is a bug.

**Comment by acerix** [comment]
I don't see any event in your demo but tried in the debug template and can see it if I drag for longer than 1 second it appears as if an external event was dropped:

https://codepen.io/acerix/pen/LYLgGGL?editors=001

It might work to use `eventReceive` as a workaround since that fires.

**Comment by quorak** [comment]
Can we fix this issue? Does anyone has a clue why this is happening? Happy to work on a patch!

**Comment by quorak** [comment]
I think I found  a proper fix to the problem here https://github.com/fullcalendar/fullcalendar-react/pull/176

**Comment by eiji03aero** [comment]
I have encountered a similar issue, and leaving some notes on how I avoided it.

TL;DR
- My team was using fullcalenar v5
- Changed the version to v6 solved the issue.

As far as I researched the issue and codebase of fullcalenar, it seemed to be the problem lays in how `optionOverrides` props are given for `CalendarDataProvider` component.
https://github.com/fullcalendar/fullcalendar-react/blob/7dfcc667665d4d434b4d25b649c73ed09a98c951/src/main.tsx#L17

Here it passes `this.props`.
The problem is that everytime this `FullCalendar` component is rendered, `optionOverrides` will become new object, thus it will trigger further recreation of objects in CalendarDataManager, which leads to recreation of context value inside `CalendarContent` component. (Sorry the memory about this mechanism part is vague already)

I tried to workaround this without making modification to source code of fullcalendar, but could not come up with any idea.

Then started taking a look at v6 codebase, and it seemed to be the problem mentioned above was cleared by little bit of refactoring, so I tried updating the version, and the bug is gone.

Hope this helps somebody.

**Comment by kaeverens** [comment]
thanks @eiji03aero 

I've added a copy of my v5 test with v6.1.8 and it does appear to be fixed as you said.

https://demo.verens.com/fullcalendar-bug/bug-v6.html

closing this.
